### DISCUSSION TITLE
Is it normal to have this high CPU utilization with a edge TPU

### DISCUSSION THREAD

**Discussion by Cytomax55** [blakeblackshear/frigate#6427] [General]
![2023-05-07_22-54](https://user-images.githubusercontent.com/39972769/236725110-86b080f8-a135-4e21-a4cc-2f65d4fd690b.png)
![2023-05-07_22-50](https://user-images.githubusercontent.com/39972769/236725090-47edc665-9f87-407f-b02d-4473cafeff1d.png)

I am using a 
Amcrest IP8M-2493EB Version 1

+++++++++++++++++
I have attached my 

Amcrest Settings

Frigate > System showing CPU utlization
++++++++++++++++

I am using the following product https://coral.ai/products/m2-accelerator-ae

stream #1 to record    
and
stream #2 to detect

++++++++++++++++++++++++++

Here my config.yaml

`mqtt:
  host: 192.168.0.10
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: user
  password: pw

detectors:
  coral_pci:
    type: edgetpu
    device: pci

go2rtc:
  streams:
    camera_41:
      - rtsp://user:pw@192.168.0.41:554/cam/realmonitor?channel=1&subtype=0
    camera_41_sub:
      - rtsp://user:pw@192.168.0.41:554/cam/realmonitor?channel=1&subtype=2
    camera_42:
      - rtsp://user:pw@192.168.0.42:554/cam/realmonitor?channel=1&subtype=0
    camera_42_sub:
      - rtsp://user:pw@192.168.0.42:554/cam/realmonitor?channel=1&subtype=2
    camera_43:
      - rtsp://user:pw@192.168.0.43:554/cam/realmonitor?channel=1&subtype=0
    camera_43_sub:
      - rtsp://user:pw@192.168.0.43:554/cam/realmonitor?channel=1&subtype=2
    

cameras:
  camera_41:
    enabled: True
    ffmpeg:
      #hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://127.0.0.1:8554/camera_41_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
        - path: rtsp://127.0.0.1:8554/camera_41
          input_args: preset-rtsp-restream
          roles:
            - record
    detect:
      width: 1280
      height: 720
      fps: 5
      enabled: True
      max_disappeared: 25
    objects:
      track:
        - person
        - dog
        - bicycle
        - cat
      filters:
        person:
          min_score: 0.4
          threshold: 0.5
          min_area: 700
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      crop: True
      height: 500
      retain:
        default: 3
    motion:
      mask:
        - 1240,22,1244,68,966,68,965,21
      threshold: 20
      contour_area: 15
    record:
      enabled: True
      retain:
        days: 2
        #mode: active_objects
      events:
        retain:
          default: 5
          mode: motion
        pre_capture: 3
        post_capture: 10`

Question #1
I am noob with ffmpeg but Why is ffmpeg being used at all since i am not transcoding anything at least not on purpose, everything is h264

Question #2
Why does it show CPU utlization so high on the camera ffmpeg

Question #3
I have a Intel Core i3-7320 @ 4x 4.1GHz that supports QuickSync, would that help with the ffmpeg CPU utilization assuming im doing everything right and i have to use ffmpeg for w/e reason

Question #4
Why is coral_pci  showing CPU % at 0 > shouldnt this be scanning the images and be higher?

Thanks in Advance

**Comment by NickM-27** [comment]
> I am noob with ffmpeg but Why is ffmpeg being used at all since i am not transcoding anything at least not on purpose, everything is h264

ffmpeg is used to take your h264 stream which is compressed and turn that in to individual frames which is what frigate runs motion and object detection on. This is why frigate recommend a GPU for hardware accelerated decoding.

ffmpeg is also used for managing recordings (though this uses minimal CPU)

> Why does it show CPU utlization so high on the camera ffmpeg

Because video decoding is an expensive task. 

> I have a Intel Core i3-7320 @ 4x 4.1GHz that supports QuickSync, would that help with the ffmpeg CPU utilization assuming im doing everything right and i have to use ffmpeg for w/e reason

Yes, you should setup vaapi 

```yaml
ffmpeg:
  hwaccel_args: preset-vaapi
```

> Why is coral_pci showing CPU % at 0 > shouldnt this be scanning the images and be higher?

This measures the CPU usage of the process managing the coral, not the usage of the coral itself.

**Reply by Cytomax55** [reply]
![2023-05-07_23-25](https://user-images.githubusercontent.com/39972769/236727226-2ae7e715-7f95-4bf1-8ad9-075d88716856.png)


cameras:
  camera_41:
    enabled: True
    ffmpeg:
      hwaccel_args: preset-vaapi


Well that seems to of solved that problem...THANK YOU very much!!!!!!
